src/plugins/ucx/ucx_backend.cpp

```diff
@@ -459,17 +488,25 @@ nixlUcxEngine::nixlUcxEngine (const nixlBackendInitParams* init_params)
         return;
     }
 
+    if (pthrOn) {
+        for (auto &uw: uws) {
+            int fd;
+            ucs_status_t ret = ucp_worker_get_efd(uw->getWorker(), &fd);
+            if (ret != UCS_OK) {
+                NIXL_ERROR << "Couldn't obtain fd for a worker, status: " << ucs_status_string(ret);
+                initErr = true;
+                return;
+            }
+
+            pollFds.push_back({fd, POLLIN, 0});
+        }
+        pollFds.push_back({pthrControlFd, POLLIN, 0});
```

Comment:
can you use `ucp_worker_signal` instead of `pthrControlFd`?

Reply:
I thought about it and I don't think I can. If ucp_worker_signal() is called during the loop that progresses workers and before that specific worker has been armed, then the signal would be ignored. After that the progress thread would sleep on poll() indefinitely while the thread that called progressThreadStop() deadlocks waiting on join().

Reply:
makes sense
probably you could check `pthrStop` right before going to poll, but this also does not look nice, so up to you

Reply:
ok